src/ar/Marker.h

```diff
@@ -0,0 +1,48 @@
+#pragma once
+
+#include <memory>
+#include <opencv2/core.hpp>
+#include <cstdint>
+
+/**
+   A class to represent the pattern that may appear on a physical AR tag, NOT a physical
+   instance of a tag itself; those are represented by the Tag class. Markers are square
```

Comment:
I'm a little unclear on the difference here. Is it just that an image may have multiple Tags corresponding to the same Marker?

Reply:
Yeah, I think in the competition they will have these triangular prisms mounted on the posts with each face being a tag that has the same marker pattern, so it could be possible that there are multiple tags corresponding to the same marker. It also seemed to make sense to me when I first introduced this class to keep the concept of the pattern on the tag and the physical tag separate in a way. Now that you mention it though, I agree the naming is a little confusing; maybe `MarkerPattern` for the pattern, keeping `Tag` for the physical occurrence?

Reply:
The name `MarkerPattern` does seem a bit clearer, good idea! Also just updating this comment so it says something like

"A class to represent an AR tag pattern that may appear in an image. Each individual instance of the pattern is represented by an instance of the Tag object; there may be multiple Tags corresponding to the same MarkerPattern."